sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2267,16 +2267,23 @@ case class GpuOverrides() extends Rule[SparkPlan] with Logging {
     if (conf.isSqlEnabled) {
       val wrap = GpuOverrides.wrapPlan(plan, conf, None)
       wrap.tagForGpu()
-      wrap.runAfterTagRules()
-      val exp = conf.explain
-      if (!exp.equalsIgnoreCase("NONE")) {
-        val explain = wrap.explain(exp.equalsIgnoreCase("ALL"))
-        if (!explain.isEmpty) {
-          logWarning(s"\n$explain")
+      val reasonsToNotReplaceEntirePlan = wrap.getReasonsNotToReplaceEntirePlan
+      if (conf.allowDisableEntirePlan && reasonsToNotReplaceEntirePlan.nonEmpty) {
+        logWarning("Can't replace any part of this plan due to: " +
```

Comment:
Should this warning be controlled by `conf.explain`? Because if it is not set then we can run into a situation where you run a command on data bricks and this warning comes out, but you don't know it is a small metadata query that runs before your actual query, which can be replaced.

Reply:
wrapped it, I only checked for !None and print it the rest of the time since it would fall into both not on gpu and all.